src/docs/arch-session.dox

```diff
@@ -0,0 +1,52 @@
+/*! @arch_page arch-session Session
+
+@section arch_session_def Definition
+After a @ref arch-connection has been established between the application and WiredTiger, the
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations are performed
+under the context of a session.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session and it is possible to create multiple sessions through the same
+connection. In fact, one connection can have multiple sessions but one session can only be
+associated with one connection. The maximum number of sessions is set through the configuration item
+\c session_max as part of the configuration string in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user sessions". WiredTiger also
+performs some internal operations such as @ref arch-eviction through self-created sessions. These
+sessions are called "internal sessions". The usage rules and guidelines for both internal sessions
+and user sessions are the same and the only difference between them is their origin of creation.
+
+@section arch_session_ops Operations
+The different operations that can be performed on a WiredTiger session are related to cursors
+handles, table operations, and transactions. You can read the complete description of each possible
+operation in the documentation related to WT_SESSION.
+
+@section arch_session_txn Transactions
+It is possible to group several operations within a session, in other words, multiple operations can
+be treated as a single atomic operation. This can be done using @ref arch-transaction. Furthermore,
+one session can hold a unique transaction at any given time and this transaction only belongs to
+that session.
+
+@section arch_session_cur Cursors
+A session can perform multiple data operations on one or several collections using multiple cursors
+(see @ref arch-cursor for more details). All the cursors associated with a session share that
+session transaction context. It is also possible to cache those cursors if required through the
+configuration string given to WT_CONNECTION::open_session or ::wiredtiger_open.
```

Comment:
Since you mentioned cached cursors and config string, maybe we should provide the config option here? The same as you did with `session_max`.

Reply:
Sounds like a good idea !

Reply:
fdcceeeef202c196459e750569e8288dc766ca49